ewatercycle/models/lisflood.py

```diff
@@ -15,165 +12,183 @@
                     str(self.forcing_dir)
                 ],
-                work_dir=str(self.work_dir),
+                work_dir=str(work_dir),
             )
         else:
             raise ValueError(
                 f"Unknown container technology in CFG: {CFG['container_engine']}"
             )
-        return Path(config_file), self.work_dir
+        return config_file, work_dir
 
     def _check_forcing(self, forcing):
         """"Check forcing argument and get path, start and end time of forcing data."""
         # TODO check if mask has same grid as forcing files,
         # if not warn users to run reindex_forcings
-        if isinstance(forcing, PathLike):
-            self.forcing_dir = forcing.expanduser().resolve()
-            self.forcing_files = dict()
-            for forcing_file in self.forcing_dir.glob('*.nc'):
-                dataset = xr.open_dataset(forcing_file)
-                # TODO check dataset was created by ESMValTool, to make sure var names are as expected
-                var_name = list(dataset.data_vars.keys())[0]
-                self.forcing_files[var_name] = forcing_file.name
-                # get start and end date of time dimension
-                # TODO converting numpy.datetime64 to datetime object is ugly, find better way
-                self.start = datetime.utcfromtimestamp(dataset.coords['time'][0].values.astype('O') / 1e9)
-                self.end = datetime.utcfromtimestamp(dataset.coords['time'][-1].values.astype('O') / 1e9)
-        elif isinstance(forcing, ForcingData):
-            # key is cmor var name and value is path to NetCDF file
-            self.forcing_files = dict()
-            data_files = list(forcing.recipe_output.values())[0].data_files
-            for data_file in data_files:
-                dataset = data_file.load_xarray()
-                var_name = list(dataset.data_vars.keys())[0]
-                self.forcing_files[var_name] = data_file.filename.name
-                self.forcing_dir = data_file.filename.parent
-                # get start and end date of time dimension
-                # TODO converting numpy.datetime64 to datetime object is ugly, find better way
-                self.start = datetime.utcfromtimestamp(dataset.coords['time'][0].values.astype('O') / 1e9)
-                self.end = datetime.utcfromtimestamp(dataset.coords['time'][-1].values.astype('O') / 1e9)
+        if isinstance(forcing, LisfloodForcing):
+            self.forcing = forcing
+            self.forcing_dir = Path(forcing.directory).expanduser().resolve()
+            # convert date_strings to datetime objects
+            self._start = get_time(forcing.start_time)
+            self._end = get_time(forcing.end_time)
         else:
             raise TypeError(
-                f"Unknown forcing type: {forcing}. Please supply either a Path or ForcingData object."
+                f"Unknown forcing type: {forcing}. Please either supply a LisfloodForcing object."
```

Comment:
There is no other way then LisfloodForcing
```suggestion
                f"Unknown forcing type: {forcing}. Please supply a LisfloodForcing object."
```